test/lib/MangroveTest.sol

```diff
@@ -408,4 +410,37 @@ contract MangroveTest is Test2, HasMgvEvents {
       assertTrue(success, "function should not revert");
     }
   }
+
+  /// creates `fold` offers in the (outbound, inbound) market with the same `wants`, `gives` and `gasreq` and with `caller` as maker
+  function densify(
+    address outbound,
+    address inbound,
+    uint wants,
+    uint gives,
+    uint gasreq,
+    uint pivotId,
+    uint fold,
+    address caller
+  ) internal {
+    if (gives == 0) {
+      return;
+    }
+    uint prov = reader.getProvision(outbound, inbound, gasreq, 0);
+    while (fold > 0) {
+      vm.prank(caller);
+      pivotId = mgv.newOffer{value: prov}(outbound, inbound, wants, gives, gasreq, 0, pivotId);
+      fold--;
+    }
+  }
+
+  /// duplicates `fold` times all offers in the `outbound, inbound` list from id `fromId` and for `lenght` offers.
+  function densify(address outbound, address inbound, uint fromId, uint length, uint fold, address caller) internal {
+    while (length > 0 && fromId != 0) {
+      MgvStructs.OfferPacked offer = mgv.offers(outbound, inbound, fromId);
+      MgvStructs.OfferDetailPacked detail = mgv.offerDetails(outbound, inbound, fromId);
+      densify(outbound, inbound, offer.wants(), offer.gives(), detail.gasreq(), fromId, fold, caller);
```

Comment:
For me, it is a bit confusing that you have 2 `densify` methods, that does different things. Maybe call one of them `innerDensify`, just to clarify the difference